change PaymentMethod to a richer PaymentDetail

joshcrews opened the pull request:
Three things are happening

1. sending in the card_type when creating a PaymentDetail is supported (like amex, visa...)
2. create_transaction arg key "payment_method" is now "payment_detail_id"
3. Payment Method is renamed Payment Detail to better fit the Rock model of a PaymentDetail (visa ending in 4242, mastercard ending in 1111) than the previous "PaymentMethod" which could only be "card" or "ach"

Points 2 and 3 are a breaking change, hince the 4.0 bump

joshcrews commented:
ok, specs pass

taylorbrooks commented on lib/rock_rms/resources/payment_detail.rb:
```diff
@@ -0,0 +1,61 @@
+  class Client
+    module PaymentDetail
+      def list_payment_details(options = {})
+        options = {'$top' => 10}
```
This overwrites the options whenever they are passed in.

taylorbrooks replied:
Perhaps it would be best to either:
a) set these as the default options
b) set the `$top` parameter if it's not already set

taylorbrooks commented on lib/rock_rms/resources/payment_detail.rb:
```diff
@@ -0,0 +1,61 @@
+          7
+        when 'mastercard'
+          8
+        when 'mastercard'
```
duplicate case?

taylorbrooks commented on lib/rock_rms/resources/payment_detail.rb:
```diff
@@ -0,0 +1,61 @@
+        Response::PaymentDetail.format(res)
+      end
+
+      def create_payment_detail(payment_type:, foreign_key: nil, card_type: nil)
```
Add support for `AccountNumberMasked` please :)

joshcrews replied:
I'm trying to add AccountNumberMasked, but the api seems to be ignoring the field. And in the API docs, I can only find this as a read-only field when you $expand FinancialTransactionDetail on FinancialTransaction

do you know anything more than this?

taylorbrooks commented:
I squashed the commits and removed the version bump.

Merged.

joshcrews commented:
Won’t not going to 4.0.0 break semver?

On April 11, 2018 at 10:30:24 AM, Taylor Brooks (notifications@github.com)
wrote:

I squashed the commits and removed the version bump.

Merged.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/taylorbrooks/rock_rms/pull/21#issuecomment-380494973>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAFWUOY41qBATJrxfQCnB2nUZuFU_MS3ks5tniGPgaJpZM4S_s4L>
.
